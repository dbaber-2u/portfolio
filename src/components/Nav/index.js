import React, { useEffect } from 'react';
import { capitalizeFirstLetter } from '../../utils/helpers';

function Nav({navigation=[], currentNavigation, setCurrentNavigation}) {
    
    

    useEffect(() => {
        document.title = capitalizeFirstLetter(currentNavigation.name);
    }, [currentNavigation]);

    return (
        <nav>
            <ul className="flex-row">
                {navigation && navigation.map((navi) => (
                    <li
                    className={`mx-1 ${currentNavigation.name === navi.name && 'navActive'}`}
                    key={navi.name}
                >
                    <span onClick={() => {
                        setCurrentNavigation(navi);
                    }}
                    >
                        {capitalizeFirstLetter(navi.name)}
                    </span>
                </li>
                ))}
            </ul>
        </nav>
    )
}

export default Nav;