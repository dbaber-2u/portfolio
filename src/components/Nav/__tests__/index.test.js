import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Nav from '..';


const currentNavigation = { "name": "About Me" };
afterEach(cleanup);



describe('Navigation component', () => {
    // baseline test
    it('renders', () => {
        render(<Nav navigation={[currentNavigation]} currentNavigation={currentNavigation} />);
    });

    // snapshot test
    it('matches snapshot', () => {
        const { asFragment } = render(<Nav currentNavigation={currentNavigation} />);

        expect(asFragment()).toMatchSnapshot();
    });
})