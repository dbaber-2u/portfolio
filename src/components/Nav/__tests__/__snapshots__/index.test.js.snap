// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component matches snapshot 1`] = `
<DocumentFragment>
  <nav>
    <ul
      class="flex-row"
    />
  </nav>
</DocumentFragment>
`;
